bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/internal/factory/ScriptedModuleHandlerFactory.java

```diff
@@ -0,0 +1,243 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.automation.module.script.rulesupport.internal.factory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Locale;
+
+import org.eclipse.smarthome.automation.Action;
+import org.eclipse.smarthome.automation.Condition;
+import org.eclipse.smarthome.automation.Module;
+import org.eclipse.smarthome.automation.Trigger;
+import org.eclipse.smarthome.automation.handler.BaseModuleHandlerFactory;
+import org.eclipse.smarthome.automation.handler.ModuleHandler;
+import org.eclipse.smarthome.automation.module.script.rulesupport.internal.handler.ScriptedHandler;
+import org.eclipse.smarthome.automation.module.script.rulesupport.internal.handler.SimpleActionHandlerWrapper;
+import org.eclipse.smarthome.automation.module.script.rulesupport.internal.handler.SimpleConditionHandlerWrapper;
+import org.eclipse.smarthome.automation.module.script.rulesupport.internal.shared.modulehandler.ScriptedActionHandlerFactory;
+import org.eclipse.smarthome.automation.module.script.rulesupport.internal.shared.modulehandler.ScriptedConditionHandlerFactory;
+import org.eclipse.smarthome.automation.module.script.rulesupport.internal.shared.modulehandler.ScriptedTriggerHandlerFactory;
+import org.eclipse.smarthome.automation.module.script.rulesupport.internal.shared.simple.SimpleActionHandler;
+import org.eclipse.smarthome.automation.module.script.rulesupport.internal.shared.simple.SimpleConditionHandler;
+import org.eclipse.smarthome.automation.type.ModuleType;
+import org.eclipse.smarthome.automation.type.ModuleTypeProvider;
+import org.eclipse.smarthome.core.common.registry.ProviderChangeListener;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ModuleHandlerFactory to provide types for "private" scripted Actions and Conditions
+ *
+ * @author Simon Merschjohann
+ *
+ */
+public class ScriptedModuleHandlerFactory extends BaseModuleHandlerFactory
+        implements ModuleTypeProvider, IScriptedModuleHandlerFactory {
+    private Logger logger = LoggerFactory.getLogger(ScriptedModuleHandlerFactory.class);
+
+    private Collection<String> types = null;
+
+    private final HashMap<String, ScriptedHandler> typesHandlers = new HashMap<>();
+    private final HashMap<String, ScriptedHandler> privateTypes = new HashMap<>();
+    private final HashMap<String, ModuleType> modulesTypes = new HashMap<>();
+
+    private int nextId = 0;
+
+    private ServiceRegistration<?> mtpReg;
+    private ServiceRegistration<?> bmhfReg;
```

Comment:
The bmhfReg is no longer being initialized. It was causing NPEs for some of my scripts.